src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -267,12 +277,38 @@ let run (conf : Scan_CLI.conf) : Exit_code.t =
                 Some (file_match_results_hook conf filtered_rules) )
           | { output_format; _ } -> (output_format, None)
         in
+        Profiler.save profiler ~name:"core_time";
         let (res : Core_runner.result) =
           Core_runner.invoke_semgrep_core
             ~respect_git_ignore:conf.targeting_conf.respect_git_ignore
             ~file_match_results_hook conf.core_runner_conf filtered_rules errors
             targets
         in
+        Profiler.save profiler ~name:"core_time";
+
+        Metrics_.add_engine_type
+          ~name:
+            (Format.asprintf "%a" Out.pp_engine_kind
+               res.Core_runner.core.engine_requested);
+
+        let filtered_matches =
```

Comment:
maybe you can put this in an helper function, like stat_rule_id_count or something, and give a nice comment
and signature.